### PR TITLE
Adjust SkipOnCoreClr for Match_ExcessPrefix test

### LINKED ISSUE #59541: System.Text.RegularExpressions.Tests.RegexMatchTests.Match_ExcessPrefix Failing in CI
Runfo Creating Tracking Issue (data being generated)

### PR DESCRIPTION
**Pull request by am11**
We wanted this test to skip on: `RuntimeConfiguration.Checked` or `RuntimeTestModes.JitMinOpts`,
but `SkipOnCoreClr` performs `RuntimeConfiguration.Checked` and `RuntimeTestModes.JitMinOpts`.

Related: #10680
Fixes: #59541

cc @stephentoub

### CODE REVIEW

File: src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs
Comment (am11) [inline src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs:1099]: @AaronRobinsonMSFT, is it safe to use:
```suggestion
            if (string.Equals(GetEnvironmentVariableValue("DOTNET_JITMinOpts"), "1", StringComparison.InvariantCulture)))
```
here?

File: src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs
Comment (am11) [inline src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs:1100]: @AaronRobinsonMSFT, is it safe to use:
```suggestion
            if (string.Equals(Environment.GetEnvironmentVariable("DOTNET_JITMinOpts"), "1", StringComparison.InvariantCulture))
```
here? Locally it works.

**Comment by stephentoub** [comment]
> but SkipOnCoreClr performs RuntimeConfiguration.Checked and RuntimeTestModes.JitMinOpts

Are there other uses of SkipInCoreClr that specify multiple conditions? Given that ConditionalXx runs the test only if all conditions pass, as the inverse it would logically make sense if SkipXx didn't run the test if any of them were an issue. So unless we're depending on this in other places,  I'd be inclined to fix the attribute instead. But we should also validate other attributes to make sure we maintain consistent behavior so that they remain predictable.
cc: @safern

**Comment by am11** [comment]
> Are there other uses of SkipOnCoreClr

There are only two other places where multiple conditions are used:
```sh
runtime % git grep 'SkipOnCoreClr.*,.*,' :/
src/libraries/System.Drawing.Common/tests/mono/System.Drawing/BitmapTests.cs:    [SkipOnCoreClr("https://github.com/dotnet/runtime/issues/37082", TestPlatforms.AnyUnix, RuntimeConfiguration.Checked)]
src/libraries/System.Runtime.Numerics/tests/BigInteger/BigIntegerToStringTests.cs:        [SkipOnCoreClr("Long running tests: https://github.com/dotnet/runtime/issues/11980", TestPlatforms.Linux, RuntimeConfiguration.Checked)]

# ruling out multi-line case
runtime % git grep SkipOnCoreClr :/ | grep -v ']'
# nothing
```
 Looks like we want `&&` effect in those two places.

**Comment by am11** [comment]
Browser_wasm_Windows failure is #59556.

File: src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs
Comment (danmoseley) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs:1100]: Is it worth moving this to a property in PlatformDetection?

Comment (AaronRobinsonMSFT) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs:1100]: @am11 Yes, that will work for the product. However, superpmi that doesn't respect the `DOTNET_` prefix yet. I need to get to that at some point /cc @BruceForstall. The pattern should be clear with the new [`clrconfignocache.h`](https://github.com/dotnet/runtime/blob/main/src/coreclr/inc/clrconfignocache.h) but it needs to happen. See https://github.com/dotnet/runtime/blob/84a622590c9cfe37544cc3753029248f7e46067e/src/coreclr/ToolBox/superpmi/superpmi/jithost.cpp#L13-L46

Comment (am11) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs:1100]: It makes sense.

After reading @stephentoub's comment, another option I was thinking of for a more permanent fix could be that we introduce something like `enum ConditionControl` with `LogicalOR`, `LogicalAnd` etc. in arcade, then change [this logic](https://github.com/dotnet/arcade/blob/3ea0d860c6973f2cbadc9e895c7ec2cbdaec4ad5/src/Microsoft.DotNet.XUnitExtensions/src/Discoverers/SkipOnCoreClrDiscoverer.cs#L29) to support both behaviors in the attribute. Other trait attributes (and their accompanying discovers in that directory) may also be able to take advantage of it. However, that kind of change would require to wait for the next Arcade update. :)

Comment (am11) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs]: Thanks @AaronRobinsonMSFT, I have updated it with `DOTNET_` prefix. Also, moved the condition to `PlatformDetection`.

Maybe we should also replace  `COMPlus_` with `DOTNET_` in SkipOnCoreClrDiscoverer (arcade repo).

Comment (AaronRobinsonMSFT) [reply src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs]: @am11 Well there are two issues here that we likely need to understand. The first is, do all of our build/ci scripts use `DOTNET_` as opposed to `COMPlus_`? I think the answer to that is "no", which means the above won't detect what it is trying to detect. The second is, if this is a feature that tests can branch on it might make more sense to bubble it up through the FeatureFlags system. I am dubious of that but it should be a conversation since the environment variable setting is so fluid.

**Comment by safern** [comment]
> So unless we're depending on this in other places, I'd be inclined to fix the attribute instead.

Maybe a better way to do this is change the `SkipOnCoreClrAttribute` to behave as the `ActiveIssueAttribute` which allows multiple attributes and that behaves as an OR. So that way you could add one attribute for `Checked` configuration and another attribute for the stress mode you want to skip the test.

https://github.com/dotnet/arcade/blob/e43b85e9b4e9ca62b89e407acdf0e56a2cbfd81a/src/Microsoft.DotNet.XUnitExtensions/src/Attributes/SkipOnCoreClrAttribute.cs#L10

`ActiveIssueAttribute` also behaves as an and for multiple parameters. If you want an OR, you add multiple attributes with different parameters.

**Comment by am11** [comment]
> Maybe a better way to do this is change the `SkipOnCoreClrAttribute` to behave as the `ActiveIssueAttribute` which allows multiple attributes

Sounds good. Question, do we want to make the change in arcade and wait for arcade update (and skip attribute) to fix #59541? Or should we take this change (initial simple version  8756c60) to unblock clean CI and then update arcade lazily?

**Comment by safern** [comment]
> Or should we take this change (initial simple version 8756c60) to unblock clean CI and then update arcade lazily?

If this is blocking PRs and making CI red, we should merge the quickest fix and then cleanup once the attribute supports multiple instances of it.

**Comment by am11** [comment]
Thanks, I have reverted it back to earlier version. I'll follow up on arcade (unless someone else beat me to it 🙂).

**Review by hoyosjs — APPROVED** [review]

**Comment by ViktorHofer** [comment]
What's the status of this PR? Is it ready to be merged? Asking as this test failure nukes our pass rate (it's 0% right now).

**Comment by ViktorHofer** [comment]
cc @krwq

**Comment by stephentoub** [comment]
I've disabled it until the right fix is available https://github.com/dotnet/runtime/blob/5d71e32a4ec09ac1d7a391f953ce0bab618c43dc/src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs#L1093.

**Comment by am11** [comment]
Ah, arcade was updated today, just noticed that it missed the fix commit. 😕
Nonetheless, I have adapted the code, after the next arcade update, will re-merged main to PR branch.

**Comment by am11** [comment]
Today's arcade update brought the updated `SkipOnCoreClrAttribute`. All green. :palm_tree:

**Comment by safern** [comment]
Thanks, @am11!

**Pull request merged**
